specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -352,6 +354,23 @@ export function QueryComboBox({
     hasTablePermission(field.relatedTable.name, 'create');
 
   const isReadOnly = React.useContext(ReadOnlyContext);
+
+  const viewButton = (isViewState: boolean): JSX.Element => (
+    <DataEntry.View
+      aria-pressed={state.type === 'ViewResourceState'}
+      className="ml-1"
+      disabled={
+        formatted?.resource === undefined ||
+        collectionRelationships === undefined
+      }
+      onClick={(): void => {
+        handleOpenRelated();
+        if (isViewState)
+          setState({ type: 'ViewResourceState', isReadOnly: true });
```

Comment:
handleOpenRelated() takes care of calling setState
in that case why do you call setState again on line 369?

perhaps you meant to add an argument for handleOpenRelated that would determine whether it would call setState with isReadOnly:true or false?

Reply:
after this change, both occurrences of `<DataEntry.View>` would be identically, so you don't need a function for this, but can assign the jsx expression to the variable directly and reuse that variable